scripts/js/lib/api/processHtml.ts

```diff
@@ -467,33 +469,21 @@ export function updateModuleHeadings($: CheerioAPI, $main: Cheerio<any>): void {
     });
 }
 
-function getApiType($dl: Cheerio<any>): ApiType | undefined {
+function getApiType($dl: Cheerio<any>): ApiObjectName | undefined {
   // Historical versions were generating properties incorrectly as methods.
   // We can fix this by looking at the modifier before the signature.
   // See https://github.com/Qiskit/documentation/issues/1352 for more information.
   if (hasPropertyModifier($dl)) {
     return "property";
   }
 
-  const apiClassNames = [
-    "type",
-    "function",
-    "class",
-    "exception",
-    "method",
-    "property",
-    "attribute",
-    "module",
-    "data",
-    "struct",
-    "enum",
-    "enumerator",
-  ] as const;
-  for (const className of apiClassNames) {
-    if ($dl.hasClass(className)) {
-      if (className === "type") return "typedef";
-      return className;
-    }
+  for (const [apiTypeName, apiType] of Object.entries(API_OBJECTS)) {
+    const className = apiType.cssSelector.split(".").pop();
+    if (!className)
+      throw new Error(
+        `'cssSelector' attribute must be of form 'dl.py.class', found '${apiType}'.`,
```

Comment:
Nit that it would be better to DRY `dl.py.class` by accessing `API_OBJECTS` (even though we don't expect it to change)

Reply:
Thanks, that's just supposed to be an example for the error message. I've updated it to be a bit clearer.